src/cli/serve/serve.js

```diff
@@ -125,11 +125,9 @@ function maybeSetRecentConfig(file, projectType, isDevMode, configs, method) {
   }
 
   try {
-    if (projectType === true) {
-      if (!existsSync(path)) {
-        writeMode('es');
-      }
-    } else {
+    if (!existsSync(path)) {
+      writeMode(projectType === true ? 'es' : projectType);
+    } else if (typeof projectType === 'string') {
```

Comment:
We only created new files when `--serverless` was called without a project type. It should be fixed now 😅